engine/src/main/java/io/camunda/zeebe/engine/processing/bpmn/event/IntermediateThrowEventProcessor.java

```diff
@@ -308,9 +315,23 @@ public boolean isSuitableForEvent(final ExecutableIntermediateThrowEvent element
     @Override
     public void onActivate(
         final ExecutableIntermediateThrowEvent element, final BpmnElementContext activating) {
+      compensationSubscriptionBehaviour.updateCompensationSubscription(activating);
```

Comment:
:x: Same as for the `EndEventProcessor`. Don't write a subscription `TRIGGERED` event if we don't have a subscription.